src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -56,6 +59,39 @@ public void init() {
     db = new FirestoreDB();
   }
 
+  /** Currently lets the front end know if a recipe is saved by a user or not. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String recipeID = request.getParameter("recipeID");
+    String token = request.getParameter("token");
+
+    UserRequestType requestType = UserRequestType.valueOf(request.getParameter("type"));
+
+    String uid = Auth.getUid(token, response);
+    if (uid == null) {
+      return;
+    }
+
+    switch (requestType) {
+      case SAVE:
+        if (recipeID == null) {
+          response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+          return;
+        }
+
+        // Call the FirestoreDB method.
+        Boolean isSavedRecipe = db.getUserProperty(uid, recipeID, User.SAVED_RECIPES_KEY);
```

Comment:
Use `Boolean[] isSavedRecipes = db.getUserProperty(uid, recipeIds, ...)` and simply return it (for checking multiple recipes).
There is no need for making null values false; the frontend can infer from a null value that a recipe has not been saved.

Reply:
See my comment above about multiple saved recipes

also I could just print null but it's honestly much more of a hassle to catch and deal with on the front end. Is there a tangible benefit to doing this on the front end rather than the back?

Reply:
I guess it doesn't really matter too much one way or another.